plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -170,40 +170,72 @@ func PrioritizeNodes(pod *api.Pod, podLister algorithm.PodLister, priorityConfig
 		return EqualPriority(pod, podLister, nodeLister)
 	}
 
+	var (
+		mu  sync.Mutex
+		err error
+	)
 	combinedScores := map[string]int{}
+	errC := make(chan error)
+
 	for _, priorityConfig := range priorityConfigs {
-		weight := priorityConfig.Weight
 		// skip the priority function if the weight is specified as 0
-		if weight == 0 {
+		if priorityConfig.Weight == 0 {
 			continue
 		}
-		priorityFunc := priorityConfig.Function
-		prioritizedList, err := priorityFunc(pod, podLister, nodeLister)
-		if err != nil {
-			return schedulerapi.HostPriorityList{}, err
-		}
-		for _, hostEntry := range prioritizedList {
-			combinedScores[hostEntry.Host] += hostEntry.Score * weight
+		go func(algorithm.PriorityConfig) {
+			weight := priorityConfig.Weight
+			priorityFunc := priorityConfig.Function
+			prioritizedList, err := priorityFunc(pod, podLister, nodeLister)
+			if err != nil {
+				errC <- err
+				return
+			}
+			for _, hostEntry := range prioritizedList {
+				mu.Lock()
+				combinedScores[hostEntry.Host] += hostEntry.Score * weight
+				mu.Unlock()
+			}
+			errC <- nil
+		}(priorityConfig)
+	}
+	// wait for all go routines to finish
+	for i := 0; i < len(priorityConfigs); i++ {
+		nerr := <-errC
+		if err == nil && nerr != nil {
+			err = nerr
```

Comment:
I think that we would like to return all errors, not just the first one.
So I would change to use WaitGroup for synchronization and instead of a errC channel use slice of errors (also locked by mu) and the after WaitGroup.Wait() just do:

if len(errorsSlice) > 0 {
  return utilerrors.NewAggregate(errorsSlice)
}
https://github.com/kubernetes/kubernetes/blob/master/pkg/util/errors/errors.go#L31